ansible_runner/config/_base.py

```diff
@@ -536,15 +537,14 @@ def wrap_args_for_containerization(self, args, execution_mode, cmdline_args):
         logger.debug(f"container engine invocation: {' '.join(new_args)}")
         return new_args
 
-    def _generate_container_auth_file(self, host, username, password):
-        token = "{}:{}".format(username, password)
+    def _generate_container_auth_dir(self, auth_data):
+        host = auth_data.get('host')
+        token = "{}:{}".format(auth_data.get('username'), auth_data.get('password'))
```

Comment:
What happens if either of these values end up being `None`? Will that fail later is a reasonable manner?

Reply:
No, it will not.

To state what the error handling would be - host, username, password are all required, provided that there is something in auth_data at all. The verify_ssl is optional, and should be `True` or `False` if it is. We are not validating it now, but if we were, these would be the rules.